groups/instance/cloud_init.tf

```diff
@@ -0,0 +1,25 @@
+data "cloudinit_config" "artifactory" {
+  count         = var.instance_count
```

Comment:
The `count` statement won't be required here.
Also you're creating `cloudinit_config` but it's not being applied to the instance